### PR TITLE
mv gov sub-module from shell into gov crate

### PR DESCRIPTION
**Pull request by tzemanovic**
## Describe your changes

- related to #1497

## Checklist before merging 
- [ ] If this PR has some consensus breaking changes, I added the corresponding `breaking::` labels
    - This will require 2 reviewers to approve the changes

### CODE REVIEW

File: crates/governance/src/finalize_block.rs [hunk 1]
```diff
@@ -0,0 +1,656 @@
...
+    }
+    Ok(None)
+}
+
+/// Count of passed and rejected proposals.
+#[derive(Default)]
+pub struct ProposalsResult {
+    passed: Vec<u64>,
+    rejected: Vec<u64>,
+}
```
Comment (Fraccaman) [inline crates/governance/src/finalize_block.rs:85]: maybe we can just return `Result<(), ..>` ? I dont think we need to return to the called which proposal passed and which didn't?

Comment (tzemanovic) [reply crates/governance/src/finalize_block.rs:85]: sounds good

Comment (tzemanovic) [reply crates/governance/src/finalize_block.rs]: done in [ae95058](https://github.com/anoma/namada/pull/3718/commits/ae950581613927fff73765ab7ed19b221d78286e)

**Comment by tzemanovic** [comment]
I've added one more refactor to get rid of the tx and vp cache being passed through

**Review by Fraccaman — APPROVED** [review]

**Comment by tzemanovic** [comment]
@mergifyio requeue

**Comment by tzemanovic** [comment]
@mergifyio requeue

**Comment by tzemanovic** [comment]
the `integration::ledger_tests::pgf_governance_proposal` keeps failing in CI but locally it passes for me :/

**Comment by brentstone** [comment]
@tzemanovic I was having the same problem with that failing test in #3691 - at some point after a rebasing on `main` it passed but I never figured out why it was failing in CI

**Comment by tzemanovic** [comment]
this one started failing after rebase - same in #3735

**Pull request merged**